### ISSUE TITLE
automatic dark theme setting

### ISSUE THREAD

**Issue by 532910** [browserpass/browserpass-extension#203]
Dark theme setting must be taken from the browser settings, rather than bother users with superfluous checkboxes (:

It's same as https://github.com/Drive4ik/simple-tab-groups/issues/502
and @Drive4ik says is really hard to implement.
Leave it open untill mozilla will provide a handy way.

**Comment by maximbaz** [comment]
Are you referring to the color scheme of the dark theme (i.e. which shade of dark gray to use as background), or are you referring to the automatic detection of whether to use light or dark theme?

The former one is not only hard to implement, but also not something I would like to do regardless, we spent a lot of time polishing the design and finding these particular colors to get a very neat UI, and I wouldn't want to give this away, sorry :)

The latter one is actually rather simple to implement by respecting not browser settings, but OS settings:

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
https://css-tricks.com/dark-modes-with-css/

But will we accept removing the ability to configure theme in the extension options, in favor of this automatic detection?

I _personally_ would be okay with automatic detection, but I fear others might not like this. @erayd what do you think?

And if we keep the theme configuration in the extension options, then I see little value in implementing an automatic theme detection - those who don't like the default dark theme will be totally fine going in the option and changing the theme, this will take a long time to pay off in developer time...

**Comment by 532910** [comment]
> Are you referring to the color scheme of the dark theme (i.e. which shade of dark gray to use as background)
> we spent a lot of time polishing the design and finding these particular colors

Noway, current dark theme looks really nice.

I'm about automatic detection of whether to use light or dark theme.

> But will we accept removing the ability to configure theme in the extension options, in favor of this automatic detection?

It's up to you, I see nothing wrong with tri-state selector system-dark-white with default to system.

**Comment by 532910** [comment]
I've found addon that respects main theme: Tampermonkey:

![image](https://user-images.githubusercontent.com/14617699/74349459-58af4900-4dc5-11ea-8a52-9bfe0ecc76a8.png)

![image](https://user-images.githubusercontent.com/14617699/74349382-3caba780-4dc5-11ea-8ad5-b17028bbf44f.png)

**Comment by erayd** [comment]
> I personally would be okay with automatic detection, but I fear others might not like this. @erayd what do you think?

My preference would be a tristate - light, dark, or auto. I personally use a light browser theme, but mostly have the browserpass theme set to dark. We already have a dropdown for it, so I don't see adding an 'auto' option as cluttering the settings at all. Having 'auto' as the default makes sense too I think; users that want something else will know what they need, and everybody else can simply ignore it.

**Comment by maximbaz** [comment]
I like three state too, I only fear this will be not trivial to implement for a very little value gain. But maybe I'm wrong, if someone wants to give it a go, please open a PR 🙂

**Comment by 532910** [comment]
can it be done with `@media -> prefers-color-scheme`css selector?

**Comment by maximbaz** [comment]
yes but since the logic has to be done in css, and not in js, I imagine it would become something like 

``` 
prefers-dark {
  all dark theme styles 
}

prefers-dark selected-light {
  all light theme styles 
}

...
```

**Comment by erayd** [comment]
What's the actual mechanism by which we can detect whether the browser is using a dark theme? Seems to me that this should be trivial to implement.

Why can't we use JavaScript as part of this? We use it everywhere else.

**Comment by maximbaz** [comment]
[media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) is the only known option to me; if we want to get the value into javascript, I think the only available option is to paint some hidden block differently using this media query, and then use `getComputedStyle` to see which style was applied.

**Comment by maximbaz** [comment]
Shipped in 3.8.0 :rocket:

**Comment by 532910** [comment]
https://github.com/browserpass/browserpass-extension/issues/202
still actual though
